src/riak_kv_qry_compiler.erl

```diff
@@ -485,23 +485,34 @@ hash_timestamp_to_quanta(QField, QSize, QUnit, QIndex, MaxSubQueries, Where) ->
                       (_, {S, E})  ->
                            {S, E}
                    end,
-    {Min, Max} = lists:foldl(GetMaxMinFun, {"", ""}, Where),
-    EffMin = case proplists:get_value(start_inclusive, Where, true) of
-                 true  -> Min;
-                 false -> Min + 1
-             end,
-    EffMax = case proplists:get_value(end_inclusive, Where, false) of
-                 true  -> Max + 1;
-                 false -> Max
-             end,
+    {Min1, Max1} = lists:foldl(GetMaxMinFun, {"", ""}, Where1),
+    %% if the start range is not inclusive then add one and remove the
+    %% start_inclusive flag. This is so that the query start key hashes to the
+    %% correct quanta when it is on the boundary since the start_inclusive flag
+    %% is not taken into account in the partition hashing. For example given a
+    %% one second quantum `mytime > 1999` should return keys with mytime greater
+    %% than 2000 but will hash to the quantum before 2000 and receive no results
+    %% from it.
+    case lists:keytake(start_inclusive, 1, Where1) of
+        {value, {start_inclusive, false}, WhereX}  ->
+            Where2 = WhereX,
+            Min2 = Min1 + 1;
+        _ ->
+            Where2 = Where1,
+            Min2 = Min1
+    end,
+    Max2 =
```

Comment:
Moving `Max` should not be generally a concern, except in an extreme case where the range includes a single time tick (which would make to the range collapse to 0, by incrementing the `Min`). It's just a note to make sure I read it right.